gossip/service.go

```diff
@@ -479,7 +479,7 @@ func (s *Service) RegisterEmitter(em *emitter.Emitter) {
 }
 
 // MakeProtocols constructs the P2P protocol definitions for `opera`.
-func MakeProtocols(svc *Service, backend *handler, disc enode.Iterator) []p2p.Protocol {
+func MakeProtocols(svc *Service, backend *handler, disc enode.Iterator) ([]p2p.Protocol, func()) {
```

Comment:
Should we name the second return parameter to document that this is a cleanup? Or is there a type definition somewhere that defines `type Cleanup func()` we could use here? Or should it be added?

Reply:
I like this idea, if it does not exist already I will add it since it will be needed for clean up of other dangling routines